oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/jvm/JVMSourceDispatcher.java

```diff
@@ -193,7 +197,29 @@ private void sendToThreadMetricProcess(String service,
         serviceInstanceJVMThread.setLiveCount(thread.getLiveCount());
         serviceInstanceJVMThread.setDaemonCount(thread.getDaemonCount());
         serviceInstanceJVMThread.setPeakCount(thread.getPeakCount());
+        serviceInstanceJVMThread.setRunnableStateThreadCount(thread.getRunnableStateThreadCount());
+        serviceInstanceJVMThread.setBlockedStateThreadCount(thread.getBlockedStateThreadCount());
+        serviceInstanceJVMThread.setWaitingStateThreadCount(thread.getWaitingStateThreadCount());
+        serviceInstanceJVMThread.setTimedWaitingStateThreadCount(thread.getTimedWaitingStateThreadCount());
```

Comment:
I think this would break compatibility for all previous java agent. We need to avoid this.

Reply:
You may use `hasField` to determine the version.

Reply:
> You may use `hasField` to determine the version.

What means `hasField`?
Read the log, I found the first metrics `Total Class Loader` is zero always.
So, What should I do?

Reply:
The compatibility test is not using the latest agent, it checks the compatibility with the previous release. I remember it uses 8.2 or 8.1 release agent. In that agent, you don't have these metrics. You should make that e2e test verifier doesn't check these added metrics.

My point was, you may need to check whether has errors here because the data OAP received in this case, they even don't report all these fields. Now, I think you could ignore my comments about `You may use hasField to determine the version.`, protobuf should have covered this case, and focus on different metrics verification in different test cases.

Reply:
@kezhenxu94 Do we have a chance to verify different metrics in the compatible case?

Reply:
Yes. @Switch-vov don’t share compatibility test class with simple case, copy `SimpleE2E` class for compatibility (`ComparE2E`) and replace https://github.com/apache/skywalking/blob/d15b67bd97cfe8386df0c61198d9f1ca2060089b/.github/workflows/e2e.compat.yaml#L61 with the `CompatE2E` class. Then in `CompatE2E` don’t verify those new metrics.

Reply:
@Switch-vov Please follow ☝️

Reply:
OK

Reply:
> @Switch-vov Please follow ☝️

I'm try to refactor SimpleE2E, CompatE2E and KafkaE2E.